crates/forge_main/src/ui.rs

```diff
@@ -837,6 +843,14 @@ impl<A: API, F: Fn() -> A> UI<A, F> {
 
         Ok(())
     }
+
+    async fn trace_user(&self) -> Result<()> {
+        let user_info = self.api.user_info().await?;
+        if let Some(login) = user_info.auth_provider_id {
+            tracker::login(login);
```

Comment:
If login needs to be called once then let's make that logic part of the tracker. Making tracker::login idempotent. Internally we can set that info in the instance of the tracker.